src/bin/light.rs

```diff
@@ -0,0 +1,367 @@
+
+    let seed = 1234; // very random, very secure seed
+
+    // Init network stack
+    let stack = &*mk_static!(
+        Stack<WifiDevice<'_, WifiStaDevice>>,
+        Stack::new(
+            wifi_interface,
+            config,
+            mk_static!(StackResources<3>, StackResources::<3>::new()),
+            seed
+        )
+    );
+
+    spawner.spawn(connection(controller)).ok();
+    spawner.spawn(net_task(&stack)).ok();
+
+    loop {
+        if stack.is_link_up() {
+            break;
+        }
+        Timer::after(Duration::from_millis(500)).await;
+    }
+
+    let base_uri;
+    println!("Waiting to get IP address...");
+    loop {
+        if let Some(config) = stack.config_v4() {
+            println!("Got IP: {}", config.address);
+            base_uri = format!("http://{}", config.address.address());
+            break;
+        }
+        Timer::after(Duration::from_millis(500)).await;
+    }
+
+    let device_id = stack.hardware_address();
+
+    let td = Thing::builder("shtc3")
+        .finish_extend()
+        .id(format!("urn:example/light/{device_id}"))
+        .base(base_uri)
+        .description("Example Thing controlling a light source")
+        .security(|builder| builder.no_sec().required().with_key("nosec_sc"))
+        .property("on", |p| {
+            p.finish_extend_data_schema()
+                .attype("OnOffProperty")
+                .title("On/Off")
+                .description("The light source is on if the property is true, off otherwise")
+                .form(|f| {
+                    f.href("/properties/on")
+                        .op(wot_td::thing::FormOperation::ReadProperty)
+                        .op(wot_td::thing::FormOperation::WriteProperty)
+                })
+                .bool()
+                .read_only()
+        })
+        .property("brightness", |p| {
+            p.finish_extend_data_schema()
+                .attype("BrightnessProperty")
+                .title("Light source brightness")
+                .description("Light source color expressed as 8bit rgb")
+                .form(|f| {
+                    f.href("/properties/brightness")
+                        .op(wot_td::thing::FormOperation::ReadProperty)
+                        .op(wot_td::thing::FormOperation::WriteProperty)
+                })
+                .number()
+                .read_only()
+        })
+        .property("color", |p| {
+            p.finish_extend_data_schema()
+                .attype("ColorProperty")
+                .title("Light source color")
+                .description("Light source color expressed as 8bit rgb")
+                .form(|f| {
+                    f.href("/properties/color")
+                        .op(wot_td::thing::FormOperation::ReadProperty)
+                        .op(wot_td::thing::FormOperation::WriteProperty)
+                })
+                .number() // TODO add a schema for rgb
```

Comment:
Should we rather set this to `object` for now? Or do you want to define a proper schema before merging this one?